testing/trino-tests/src/test/java/io/trino/execution/EventsCollector.java

```diff
@@ -57,12 +64,16 @@ public synchronized void reset(int numEvents)
     public void waitForEvents(int timeoutSeconds)
             throws InterruptedException
     {
+        CountDownLatch eventsLatch;
+        synchronized (this) {
+            eventsLatch = this.eventsLatch;
```

Comment:
This could major problem, since the reset could happen on a separate thread, and then you could wait on the old event latch

Reply:
How would that help? You can still reset `eventsLatch` to new instance just after we get out of this synchronized section, and then you are still waiting on an old one.
It looks to me that what we need to do is to ensure that the old latch is fully released in the `reset` method before we switch it to new one.

Reply:
Or is this just a memory barier to ensure variable visibility?

Reply:
nit: imo worth adding the explanation as a comment in the code for future readers